File: Algorithm/Other/at coder/Other/A38/A38.py
def main():
    import sys
    input = sys.stdin.read
    data = input().split()

    D = int(data[0])
    N = int(data[1])
    
    # 各日の最大労働時間を24時間で初期化
    max_hours = [24] * D

    # 制約に基づいて制限
    index = 2
    for _ in range(N):
        L = int(data[index])
        R = int(data[index + 1])
        H = int(data[index + 2])
        index += 3
        for day in range(L - 1, R):
            max_hours[day] = min(max_hours[day], H)

    # 合計を出力
    print(sum(max_hours))

if __name__ == "__main__":
    main()

File: Algorithm/Other/at coder/Other/A38/A38.js
const fs = require('fs');

// 標準入力から読み込み
const input = fs.readFileSync('/dev/stdin', 'utf8').trim().split('\n');
const [D, N] = input[0].split(' ').map(Number);

// 各日ごとに最大で働ける時間（初期値は24）
const maxHours = Array(D).fill(24);

// 制約を反映
for (let i = 1; i <= N; i++) {
    const [L, R, H] = input[i].split(' ').map(Number);
    for (let day = L - 1; day <= R - 1; day++) {
        maxHours[day] = Math.min(maxHours[day], H);
    }
}

// 最大労働時間の合計を出力
const totalMaxHours = maxHours.reduce((sum, h) => sum + h, 0);
console.log(totalMaxHours);
